source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -182,6 +196,30 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
         }
     }
 
+    // Replace getElement(x, i) with, y = store(x); p = getElementPtr(y, i); load(p)
+    // SPIR-V has no support for dynamic indexing into values like we do.
+    // It may be advantageous however to do this further up the pipeline
+    void processGetElement(IRGetElement* inst)
```

Comment:
This is unfortunate, we could probably use an earlier pass to de-SSA composite typed values.

Reply:
definitely, a more holistic pass could generate much better code in the case of several indexed reads.